packages/helper-regex-builder/index.js

```diff
@@ -1,21 +1,33 @@
 import escapeRegexString from 'escape-regex-string';
 
-function regexBuilder(key, value, groupKey) {
+function createRegEx(pattern, globalSearch) {
+  if (globalSearch) {
+    return new RegExp(pattern, 'g');
+  }
+
+  return new RegExp(pattern);
+}
+
+function regexBuilder(key, value, groupKey, globalSearch) {
   const regexKey = escapeRegexString(key);
   const keyField = groupKey ? `("${regexKey}")` : `"${regexKey}"`;
   if (Object.prototype.toString.call(value) !== '[object String]') {
-    return new RegExp(`${keyField}`, 'g');
+    return createRegEx(`${keyField}`, globalSearch);
   }
 
   const regexValue = escapeRegexString(value);
   const valueField = `"(${regexValue})"`;
-  return new RegExp(`${keyField}\\s*:\\s*${valueField}`, 'g');
+  return createRegEx(`${keyField}\\s*:\\s*${valueField}`, globalSearch);
 }
 
 export function jsonRegExKeyValue(key, value) {
-  return regexBuilder(key, value, true);
+  return regexBuilder(key, value, true, true);
+}
+
+export function jsonRegExSingleKeyValue(key, value) {
+  return regexBuilder(key, value, true, false);
```

Comment:
Personally I would prefer adding another argument to `jsonRegExKeyValue` instead of introducing another function. Basically how it was until https://github.com/OctoLinker/OctoLinker/commit/93d85161cd3eacac9ad467dcdd55075a2566a224#diff-b0829a6c903b7a1200ba3d8cfbf4f331

Reply:
That's actually the exact setup I had but I switched to this thinking it'd be easier to maintain. Now it makes sense as to why  there's unit tests with a 3rd parameter for those functions even though they only have 2.

Reply:
Up to you, I see value in both approaches ... at the end it doesn't really matter 😉 

Regarding the tests, in #774 I had to change quite a lot so it sneaked through ... sorry for the confusion